lib/player_input.rb

```diff
@@ -0,0 +1,51 @@
+class PlayerInput
+  attr_reader :input, :output, :validated_coordinates, :entered_coordinates
+
+  def initialize(input:, output:)
+    @input = input
+    @output = output
+  end
+
+  def coordinates
+    until validated_coordinates
+      get_and_validate_input
+    end
+    validated_coordinates
+  end
+
+  private
+
+  def get_and_validate_input
+    print_prompt
+    get_input
+    unless validate_input
+      print_error
+    end
+  end
+
+  def get_input
+    @entered_coordinates = input.gets.chomp
+  end
+
+  def validate_input
+    if valid_coordinates.include?(entered_coordinates)
+      @validated_coordinates = entered_coordinates
+    end
+  end
+
```

Comment:
These printing related methods hold together on their own, but not with the rest of the methods. I see that because of the names of the methods and also their main concern being output.

Reply:
Extracted another view object in d733979. I've started to feel a pull to having these view objects be injected, instead of the output stream itself, but I'm not sure I'm ready to make that jump just yet.